File: src/main/java/com/fitness/application/model/Goal.java

package com.fitness.application.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "goals")
public class Goal {
    @Id
    private String id;
    private String userId;
    private String title;
    private String description;
    private double target;
    private double current;
    private String type; // weight, strength, endurance, habit, custom
    private String status; // in-progress, completed, not-started
    private LocalDate deadline;
    private LocalDate createdAt;
    private int progress; // 0-100
}

File: src/types/goal.ts

export interface Goal {
  id: string;
  title: string;
  description?: string;
  target: number;
  current: number;
  type: GoalType;
  status: GoalStatus;
  progress: number;
  deadline?: string;
}

export type GoalType = 'weight' | 'workout' | 'steps' | 'distance' | 'calories' | 'custom' | 'strength' | 'endurance' | 'habit';
export type GoalStatus = 'in-progress' | 'completed' | 'missed' | 'not-started';

export interface GoalFormData {
  title: string;
  description?: string;
  type: GoalType;
  target: number;
  current?: number;
  deadline?: string;
}
